interview-questions.md

```diff
@@ -1,62 +1,76 @@
-# ASSESSMENT 2: Interview Practice Questions
+April Rieger - # ASSESSMENT 2: Interview Practice Questions
 
 Answer the following questions. First, without external resources. Challenge yourself to answer from memory. Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn.
 
 1. Why would you use the method super()?
 
-  Your answer:
+  Your answer: We are able to inherit from the parent
 
-  Researched answer:
+  Researched answer: The super() method refers to the parent class. By calling the super() method in the constructor method, we call the parent's constructor method and gets access to the parent's properties and methods. https://www.w3schools.com/js/js_classes.asp
 
 
 
 2. What is a virtual DOM?
 
-  Your answer:
+  Your answer: I don't know the definition of virtual DOM but DOM stands for the Document Object Model which n interface, so taking a stab that this is a virtual interface.
 
-  Researched answer:
+  Researched answer: In React, for every DOM object, there is a corresponding “virtual DOM object.” A virtual DOM object is a representation of a DOM object, like a lightweight copy.
+
+  A virtual DOM object has the same properties as a real DOM object, but it lacks the real thing’s power to directly change what’s on the screen.
+
+  Manipulating the DOM is slow. Manipulating the virtual DOM is much faster, because nothing gets drawn onscreen. Think of manipulating the virtual DOM as editing a blueprint, as opposed to moving rooms in an actual house.
+
+  https://www.codecademy.com/articles/react-virtual-dom
 
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
-  Your answer:
+  Your answer: You will need a function that returns the JSX value in the H1 tags (see example below).
+  class Function = props => {
+      <div><h1>Hello World</h1></div>
+    }
+  }
+  export default App
```

Comment:
syntax is a little off here but i see what you're going for

Reply:
We would need a render, return and a React.Fragment or <></> its all starting to make more sense each day :) YAY!